Feat: Calculate code coverage in getRecipientsOfQuestion

levi-tating opened the pull request:
This adds lines to getRecipientsOfQuestion that store which branches were reached during the lifetime of an instance of the FeedbackQuestionsLogic class.

ali-shahwali committed:
feat: Resolved issue #10

Added 2 new test to improve the coverage of the method `FeedbackQuestionAttributes::equals`, one that tests null cases and one that tests negative cases. Should resolve issue #10.

Hashtagsmile committed:
Fix: Added two tests to achieve 100% branch coverage in validateQuestionDetails

Penguinification committed:
Feat: Improve populateFieldsToGenerateInQuestion code coverage

This adds four new unit tests to FeedbackQuestionsLogicTest that
increase the branch coverage of the test suite to 100% according to
manual instrumentation and 96% according to JaCoCo.

Fixes #14

Hashtagsmile committed:
Merge pull request #13 from Penguinification/calculate-coverage-validateQuestionDetails-only-tests

Fix: Added two tests to achieve 100% branch coverage in validateQuest…

Hashtagsmile committed:
Merge pull request #15 from Penguinification/improve-code-coverage-populate-fields-to-generate-in-question

Feat: Improve populateFieldsToGenerateInQuestion code coverage

Penguinification committed:
Merge pull request #11 from Penguinification/issue/improve-coverage-10

feat: Resolved issue #10

Hashtagsmile committed:
Feat: Refactored the validateQuestionDetails

Hashtagsmile committed:
Feat: Tests for improved branch coverage for shouldChangesRequireResponseDeletion

levi-tating committed:
trrack coverage in getRecipientsOfQuestion

levi-tating committed:
added test for reached branches

levi-tating committed:
changed number of totalBranches

Penguinification commented:
This should be merged into code-coverage, not master. Looks good otherwise!

Penguinification commented:
The easiest way to solve this merge conflict would be to keep both changes, although this might not be the prettiest solution.

Penguinification committed:
Refactor populateFieldsToGenerateInQuestion

This splits the populateFieldsToGenerateInQuestion into four methods
with lower cyclomatic complexity. These functions are
populateFieldsToGenerateInQuestion, populateOptionList, getStudentList
and getTeamList with CC:s of 5, 11, 3 and 3, respectively.
Compared to the previous CC of 19,
populateFieldsToGenerateInQuestion has had its complexity reduced by
74%. The most complex of the new methods, populateOptionList, also has
a CC that is 42% lower than the original method.

Fixes #26

ali-shahwali committed:
refactor: Resolved issue #21 (#23)

Refactored the method `FeedbackResponsesLogic::isResponseVisibleForUser` to reduce the cyclomatic complexity, it is now 14 from 29. Approximately a 50% reduction.

ali-shahwali committed:
feat: Resolved issue #24 (#25)

Added additional tests for `FeedbackQuestionAttributes::equals` to improve coverage. Also refactored old tests from #10 for better modularity.

levi-tating committed:
Merge branch 'master' of https://github.com/Penguinification/dd2480-teammates into calculate-coverage-getRecipientsOfQuestion

Penguinification committed:
Merge pull request #27 from Penguinification/refactor-populate-fields-to-generate-in-question

Feat: Refactor populateFieldsToGenerateInQuestion

levi-tating committed:
Merge branch 'master' of https://github.com/Penguinification/dd2480-teammates into calculate-coverage-getRecipientsOfQuestion

Hashtagsmile committed:
Merge pull request #19 from Penguinification/17-refactoring-feedbackmsqquestiondetailsvalidatequestiondetails

Feat: Refactored the validateQuestionDetails

Hashtagsmile committed:
Fix: Deleted comments not needed

Hashtagsmile committed:
Merge pull request #20 from Penguinification/18-improve-branch-coverage-for-feedbackmsqquestiondetailsshouldchangesrequireresponsedeletion

Feat: Tests for improved branch coverage for shouldChangesRequireResp…

Penguinification committed:
Add link to report

This adds a link to the report to the README.md.

Fixes #28

Penguinification committed:
Merge pull request #29 from Penguinification/add-report

Add link to report

levi-tating committed:
Merge branch 'master' of https://github.com/Penguinification/dd2480-teammates into calculate-coverage-getRecipientsOfQuestion

levi-tating committed:
refactored getRecipientsOfQuestion
